a3p-integration/proposals/a:upgrade-next/priceFeed.test.js

```diff
@@ -0,0 +1,155 @@
+/* global process */
+
+import test from 'ava';
+
+import {
+  agd,
+  agops,
+  executeOffer,
+  getVatDetails,
+  GOV1ADDR,
+  GOV2ADDR,
+  GOV3ADDR,
+  newOfferId,
+} from '@agoric/synthetic-chain';
+
+const ORACLE_ADDRESSES = [GOV1ADDR, GOV2ADDR, GOV3ADDR];
+
+const getOracleInstance = async price => {
+  const instanceRec = await agd.query(
+    'vstorage',
+    'data',
+    '--output',
+    'json',
+    `published.agoricNames.instance`,
+  );
+
+  // agd query -o json  vstorage data published.agoricNames.instance
+  //    |& jq '.value | fromjson | .values[1] | fromjson | .body[1:]
+  //    | fromjson | .[-2] '
+
+  const value = JSON.parse(instanceRec.value);
+  // XXX I don't know why value.values[-1] doesn't work.
+  const body = JSON.parse(value.values[value.values.length - 1]);
+
+  const bodyTruncated = JSON.parse(body.body.substring(1));
+  const slots = body.slots;
+
+  for (const [k, v] of bodyTruncated.entries()) {
+    if (v[0] === `${price}-USD price feed`) {
+      // console.log(`PFt `, price, slots[k]);
+      return slots[k];
+    }
+  }
```

Comment:
this is a little opaque. consider,

```suggestion
  const feeds = JSON.parse(body.body.substring(1));

  const feedName = `${price}-USD price feed`;
  const key = Object.keys(feeds).find(k => feeds[k][0] === feedName);
  
  if (key) {
    return feeds.slots[key];
  }
```

Reply:
Thanks, that's much clearer.